pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -290,6 +295,11 @@ let
     ] ++ optionals (python.pythonAtLeast "3.3") [
       # Optionally enforce PEP420 for python3
       pythonNamespacesHook
+    ] ++ lib.optionals (
+      isTrue (attrs.pythonRelaxDeps or false) || isNonEmptyList (attrs.pythonRelaxDeps or []) ||
+      isTrue (attrs.pythonRemoveDeps or false) || isNonEmptyList (attrs.pythonRemoveDeps or [])
```

Comment:
Is this expression equivalent to this one?

```nix
attrs?pythonRelaxDeps || attrs?pythonRemoveDeps
```

Reply:
I believe the `?` operator only checks for the existence of an attribute: [builtins.hasAttr](https://nix.dev/manual/nix/2.22/language/builtins#builtins-hasAttr)
Goal was to test if the value is a boolean (and true) or a list with items in it.
It shouldn't add the hook for invalid values (string/int/null/etc).

Reply:
We have no type to prevent them (string/int/null/etc), but such values should not be set in the first place.
Since it is not a heavy hook, adding it by mistake is not so bad, IMO.